great_expectations/core/usage_statistics/anonymizers/checkpoint_run_anonymizer.py

```diff
@@ -0,0 +1,368 @@
+import datetime
+import logging
+from numbers import Number
+from typing import Any, Dict, List, Optional, Union
+
+import great_expectations.exceptions as ge_exceptions
+from great_expectations.checkpoint.util import get_substituted_validation_dict
+from great_expectations.core import RunIdentifier
+from great_expectations.core.batch import BatchRequest, get_batch_request_dict
+from great_expectations.core.usage_statistics.anonymizers.action_anonymizer import (
+    ActionAnonymizer,
+)
+from great_expectations.core.usage_statistics.anonymizers.anonymizer import Anonymizer
+from great_expectations.core.usage_statistics.anonymizers.batch_request_anonymizer import (
+    BatchRequestAnonymizer,
+)
+from great_expectations.core.usage_statistics.anonymizers.types.base import (
+    CHECKPOINT_OPTIONAL_TOP_LEVEL_KEYS,
+)
+from great_expectations.core.util import get_datetime_string_from_strftime_format
+from great_expectations.data_context.types.base import CheckpointConfig
+from great_expectations.util import deep_filter_properties_iterable
+
+logger = logging.getLogger(__name__)
+
+
+class CheckpointRunAnonymizer(Anonymizer):
+    def __init__(self, salt=None):
+        super().__init__(salt=salt)
+
+        self._salt = salt
+
+    # noinspection PyUnusedLocal
+    def anonymize_checkpoint_run(self, *args, **kwargs) -> Dict[str, List[str]]:
+        """
+        Traverse the entire Checkpoint configuration structure (as per its formal, validated Marshmallow schema) and
+        anonymize every field that can be customized by a user (public fields are recorded as their original names).
+        """
+        batch_request_anonymizer: BatchRequestAnonymizer = BatchRequestAnonymizer(
+            self._salt
+        )
+        action_anonymizer: ActionAnonymizer = ActionAnonymizer(self._salt)
+
+        attribute_name: str
+        attribute_value: Optional[Union[str, dict]]
+        validation_obj: dict
+        action_config_dict: dict
+        action_name: str
+        action_obj: dict
+
+        checkpoint_optional_top_level_keys: List[str] = []
+
+        name: Optional[str] = kwargs.get("name")
+        anonymized_name: Optional[str]
+        if name is None:
+            anonymized_name = None
+        else:
+            anonymized_name = self.anonymize(name)
```

Comment:
@anthonyburdi I would be hesitant to modify the base `Anonymizer` class, since it has existed for a couple of years now.  However, if you think it is safe, I am happy to do this -- how about in a quick follow-on PR (along with other potential similar improvements)?  Thanks.

Reply:
For sure - separate PR makes sense. Also this is a totally optional suggestion. If it's more trouble than it's worth, or you have other higher priority work then ditch it :)